src/kafka-connect/create-connector/create-connector.controller.js

```diff
@@ -31,118 +31,171 @@ angularAPP.controller('CreateConnectorCtrl', function ($scope, $rootScope, $http
-                 }
-               });
+              name: "",
+              config: {}
+            };
+            var errorConfigs = [];
 
+            flatValuesArray = $scope.formValuesPerSection.split('\n');
             config = NewConnectorFactory.getJSONConfigFlat(flatValuesArray);
-            var classname = $scope.connector.class;
+
+            // Make sure the 'classname' is a valid one - as it can crash the connect services
+            var classname = flatValuesArray.find(function (p) {
+                 return (p.indexOf("connector.class=") == 0)
+            }).split('connector.class=').join('');
+            if (classname != $scope.connector.class) {
+                console.log("error in classname -> " + classname);
+                var errors = { errors : [ 'Classname "' + $scope.connector.class + '" is not defined' ] };
+                errorConfigs.push(errors);
+            }
+
+            //console.log("Error configs -> " + errorConfigs);
+            //console.log(errorConfigs.length == 0);
+
+            if(errorConfigs == 0) {
+                $scope.validConfig = constants.VIEW_MESSAGE_CONNECTOR_VALID;
+                $scope.curlCommand = NewConnectorFactory.getCurlCommand(flatValuesArray);
+            }
+            $scope.errorConfigs = errorConfigs;
+
             //STEP 1: Validate
             var validateConfigPromise = KafkaConnectFactory.validateConnectorConfig(classname, config);
             validateConfigPromise.then(
                 function success(data) {
-                  $log.info('Total validation errors => ' + data.error_count);
+                  $log.info('Total validation errors from API => ' + data.error_count);
                   //STEP 2: Get errors if any
-                  var errorConfigs = [];
-                  $scope.validConfig = ''
+                  $scope.validConfig = '';
+                  var validConnectorConfigKeys = [];
+                  var requiredConfigKeys = [];
                   angular.forEach(data.configs, function (config) {
+                    //console.log("c-> " + config.value.name);
+                    //console.log(config);
                     if (config.value.errors.length > 0) {
                         errorConfigs.push(config.value);
                         $log.info(config.value.name + ' : ' + config.value.errors[0]);
                     }
+                    //console.log("config.value -> ");
+                    if ( (config.definition.required == true) || (config.value.name.startsWith("topic")) ){
+                      requiredConfigKeys.push(config.value.name);
+                    }
+                    console.log("Required/compulsory config keys: " + requiredConfigKeys);
+                    validConnectorConfigKeys.push(config.value.name);
+                  });
+                  //console.log("validConnectorConfigKeys -> " + validConnectorConfigKeys);
+                  angular.forEach(flatValuesArray, function (propertyLine) {
```

Comment:
3 loops to check the same thing with so many conditionals makes it complicated.
I suggest break it to a `validate` private function that performs the steps, put some meaningful function names because there is so much indexOf logic ie. getProperty(), or propertyExists(), and wrap the logic to be readable. 
Also if you break into smaller functions you may consider use a bit of functional programming ie list.filter(!propertyExists).map(toErrors) etc